src/CommunityToolkit.Aspire.Hosting.Minio/MinioBuilderExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Aspire.Hosting.ApplicationModel;
+using CommunityToolkit.Aspire.Hosting.Minio;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding MiniO resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class MinioBuilderExtensions
+{
+    private const string RootUserEnvVarName = "MINIO_ROOT_USER";
+    private const string RootPasswordEnvVarName = "MINIO_ROOT_PASSWORD";
+
+    /// <summary>
+    /// Adds a MiniO container to the application model. The default image is "minio/minio" and the tag is "latest".
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="minioConsolePort">The host port for MinioO Admin.</param>
+    /// <param name="minioPort">The host port for MiniO.</param>
+    /// <param name="rootUser">The root user for the MiniO server.</param>
+    /// <param name="rootPassword">The password for the MiniO root user.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MinioContainerResource}"/>.</returns>
+    public static IResourceBuilder<MinioContainerResource> AddMinioContainer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource>? rootUser = null,
+        IResourceBuilder<ParameterResource>? rootPassword = null,
+        int minioPort = 9000,
```

Comment:
```suggestion
        int? minioPort = null,
```

Expose a WithHostPort, WithPassword and WithUsername

Reply:
Thanks!

I was looking at the integrations in this repo, but this got me looking at a Postgres integration at the main Aspire repo, and I've tweaked this and few things to the better